secretary.py

```diff
@@ -306,20 +306,30 @@ def _prepare_template_tags(self, xml_document):
             parent.removeChild(discard)
 
 
-    def _unescape_entities(self, xml_text):
-        # unescape XML entities gt and lt
+    @staticmethod
+    def _unescape_entities(xml_text):
+        """
+        Strips tags of the form <text:span ...> from inside Jinja elements
+        and unescapes HTML codes for >, <, & and "
+        """
         unescape_rules = {
+            r'(?is)({[{|%].*)(<.?text:s.*?>)(.*[%|}]})': r'\1\3',
```

Comment:
Actually `text:s.?` will match more than one character. A non-greedily match will only match the first char after **`s`**. So if you need to escape `text:space` using `r'text:s.*?'` you will be only matching **`text:sp`**.

This the reason why your version is not passing automated tests.